src/JsonSchema/Uri/Retrievers/FileGetContents.php

```diff
@@ -14,51 +14,73 @@
 
 /**
  * Tries to retrieve JSON schemas from a URI using file_get_contents()
- * 
- * @author Sander Coolen <sander@jibber.nl> 
+ *
+ * @author Sander Coolen <sander@jibber.nl>
  */
 class FileGetContents extends AbstractRetriever
 {
     protected $messageBody;
-    
+
     /**
      * {@inheritDoc}
      * @see \JsonSchema\Uri\Retrievers\UriRetrieverInterface::retrieve()
      */
     public function retrieve($uri)
     {
-        $context = stream_context_create(array(
-            'http' => array(
+        $context = stream_context_create([
+            'http' => [
                 'method' => 'GET',
-                'header' => "Accept: " . Validator::SCHEMA_MEDIA_TYPE
-            )));
+                'header' => "Accept: ".Validator::SCHEMA_MEDIA_TYPE,
+            ],
+        ]);
 
-        set_error_handler(function() use ($uri) {
-            throw new ResourceNotFoundException('JSON schema not found at ' . $uri);
+        $error = null;
+        set_error_handler(function (
+            $errno,
+            $errstr,
+            $errfile,
+            $errline,
+            array $errcontext
+        ) use (&$error) {
+            $error = [
+                $errno,
+                $errstr,
+                $errfile,
+                $errline,
+                $errcontext,
+            ];
         });
         $response = file_get_contents($uri);
         restore_error_handler();
 
+        if ($error) {
+            list (, $errstr, $errfile, $errline) = $error;
+            $message = sprintf('%s on line %s in %s', $errstr, $errline,
```

Comment:
Almost everything in the proposed message template is already present in the stacktrace. $errstr is the only value you need here.